src/infrastructure/utils/ZodValidatorMiddleware.ts

```
@@ -0,0 +1,166 @@
+  },
+>(
+  schemas: T,
+) => {
+  return validator("zod", (value, c) => {
+    const errors: Record<string, string> = {};
+    let hasErrors = false;
+
+    if (schemas.params && value.params) {
+      try {
+        schemas.params.parse(value.params);
+      } catch (e: any) {
```

Biome (1.9.4)

[error] Unexpected any. Specify a different type.

any disables many type checking rules. Its use should be avoided.

(lint/suspicious/noExplicitAny)